config/v1/types_authentication.go

```diff
@@ -159,3 +175,130 @@ const (
 	// KubeConfigKey is the key for the kube config file data in a secret
 	KubeConfigKey = "kubeConfig"
 )
+
+type OIDCProvider struct {
```

Comment:
Is this the right scope for configuring a clientID and reference to a clientSecret for the console

Pros:
1. keeps configuration close together from a user perspective
2. will allow different clientIDs for different oidc providers for console

Cons?

Reply:
Each client for each application should be different, or should be provided with means to request tokens with different audiences (e.g. the `resource` parameter). That means that should we choose to configure client parameters via this API, we need to make it configurable for a number of clients. That can get messy, but is still doable.